cogs/displayinfo.py

```diff
@@ -33,85 +33,130 @@ class DisplayInfo(commands.Cog):
+            for member in guild.members:
+                if member.status == discord.Status.online and member.activity:
+                    self.members_online.add(member)
+                    after_string_id = str(member.id)
+                    if after_string_id not in self.timers:
+                        self.timers[after_string_id] = {} # If the member is not already tracked, add them
+                    if self.current_date not in self.timers[after_string_id]:
+                        self.timers[after_string_id][self.current_date] = {} # If the current date is not present in the member's dictionary, add it
+                    if member.activity.name not in self.timers[after_string_id][self.current_date]: # If the member's current activity/game/status is not present, add it with initial values
+                        self.timers[after_string_id][self.current_date][member.activity.name] = {
+                            "totalTime" : 0, 
+                            "startTime" : 0,
+                        }
+                    self.timers[after_string_id][self.current_date][member.activity.name]["startTime"] = datetime.now() # Record the start time of the member's current activity
+
+        print(self.members_online)
+
+        if len(self.members_online):
+            self.timed_task = asyncio.create_task(self.timed_update()) #start timed_update once bot goes online if a member is online
+
+
         
     # The 'mem_before' parameter represents the member's previous state, and 'mem_after' represents the current state.
     @commands.Cog.listener()
     async def on_member_update(self, mem_before:discord.Member, mem_after:discord.Member):
+        self.current_guild_name = mem_before.guild.name
+        
         if mem_before.activity and mem_before.activity == mem_after.activity:
             return
-    
-        if self.current_guild_name != mem_before.guild.name:
-            if mem_before.guild.name not in db.list_collection_names():
-                newCollection = db[mem_before.guild.name]  # If the collection doesn't exist, create a new one for the guild/server
-                self.current_guild_name = mem_before.guild.name
-                self.current_collection = newCollection               
-                newCollection.insert_one({}) # Create a new blank document in the new Collection
-                self.timers = self.current_collection.find_one() # Load blank document from the MongoDB collection
-            else:
-                self.current_collection = db[mem_before.guild.name] # If the collection exists, get the existing collection for the guild
-                self.current_guild_name = mem_before.guild.name
-                self.timers = self.current_collection.find_one() # Load existing document from the MongoDB collection
+
+        if mem_after.status == discord.Status.online and mem_after.activity: #if member is online and in an activity
+            self.members_online.add(mem_after)
```

Comment:
should we keep track of the member object in the set or just their id? we should be able to use discord.py to get a member object from the user id right? storing full member objects will be a lot of memory.